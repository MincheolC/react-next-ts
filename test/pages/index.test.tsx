import React from 'react'
import { render } from '../testUtils'
import MainPage from '../../pages/index'

jest.mock('next/image', () => ({ src }) => <img src={src} />)

describe('Main page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<MainPage />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
