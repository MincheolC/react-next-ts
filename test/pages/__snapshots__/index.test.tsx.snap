// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-4 makeStyles-appMain-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorInherit makeStyles-appBar-5 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolBar-6 MuiToolbar-gutters"
      >
        <div
          class="MuiBox-root MuiBox-root-10 makeStyles-appTitle-7"
        >
          CHARLES DEV
        </div>
        <div
          class="MuiBox-root MuiBox-root-11 makeStyles-menuContainer-8"
        >
          <a
            class="makeStyles-menu-9"
            href="/"
          >
            REACT
          </a>
          <a
            class="makeStyles-menu-9"
            href="/css"
          >
            CSS
          </a>
          <a
            class="makeStyles-menu-9"
            href="/"
          >
            FLUTTER
          </a>
          <a
            class="makeStyles-menu-9"
            href="/"
          >
            TYPESCRIPT
          </a>
          <a
            data-testid="myGithub"
            href="https://github.com/MincheolC"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              style="font-size: 30px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
              />
            </svg>
          </a>
          <a
            data-testid="myNotion"
            href="https://www.notion.so/Charles-Dev-b2c65d1ae5ee4df99d4e24abe37eb38a"
            target="_blank"
          >
            <img
              src="/images/notion-icon-96-black.png"
            />
          </a>
        </div>
      </div>
    </header>
    <div
      class="MuiBox-root MuiBox-root-12 makeStyles-mainContent-2"
    >
      <div
        class="MuiBox-root MuiBox-root-13 makeStyles-mainContainer-3"
      >
        <div
          class="MuiBox-root MuiBox-root-17 makeStyles-mainBanner-14"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root makeStyles-mainPost-15 MuiGrid-item MuiGrid-grid-xs-8"
            >
              <h2
                class="MuiTypography-root makeStyles-title-18 makeStyles-title-19 MuiTypography-h2"
              />
              <div
                class="MuiBox-root MuiBox-root-26 makeStyles-container-20"
              >
                <img
                  src="https://picsum.photos/1200/800"
                />
                <div
                  class="MuiBox-root MuiBox-root-27 makeStyles-titleContiner-21"
                >
                  <div
                    class="MuiBox-root MuiBox-root-28 makeStyles-title-22"
                  >
                    Three Things in Life That Aren’t Worth The Effort
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-29 makeStyles-content-23"
                  >
                    To be more efficient and happy, cut the waste and damaging activities from your life.
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-30 makeStyles-createdAt-24"
                  >
                    Dec 24, 2020
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-31 makeStyles-tags-25"
                  >
                    <span
                      class="MuiBox-root MuiBox-root-33 makeStyles-tag-32"
                    >
                      #Life
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
              style="padding-left: 3%;"
            >
              <h2
                class="MuiTypography-root makeStyles-title-18 makeStyles-title-34 MuiTypography-h2"
              >
                Latest Post
              </h2>
              <div>
                <div
                  class="MuiBox-root MuiBox-root-41 makeStyles-container-35"
                >
                  <div
                    class="MuiBox-root MuiBox-root-42 makeStyles-titleContiner-36"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-43 makeStyles-title-37"
                    >
                      [React Props Cheatsheet: 10 Patterns You Should Know
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-44 makeStyles-createdAt-39"
                    >
                      Mar 14, 2021
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-45 makeStyles-tags-40"
                    >
                      <span
                        class="MuiBox-root MuiBox-root-46 makeStyles-tag-32"
                      >
                        #React
                      </span>
                      <span
                        class="MuiBox-root MuiBox-root-47 makeStyles-tag-32"
                      >
                        #Front-end
                      </span>
                    </div>
                  </div>
                  <img
                    src="https://picsum.photos/403/403"
                  />
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-middle"
                />
              </div>
              <div>
                <div
                  class="MuiBox-root MuiBox-root-48 makeStyles-container-35"
                >
                  <div
                    class="MuiBox-root MuiBox-root-49 makeStyles-titleContiner-36"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-50 makeStyles-title-37"
                    >
                      [React Props Cheatsheet: 10 Patterns You Should Know
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-51 makeStyles-createdAt-39"
                    >
                      Mar 11, 2021
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-52 makeStyles-tags-40"
                    >
                      <span
                        class="MuiBox-root MuiBox-root-53 makeStyles-tag-32"
                      >
                        #React
                      </span>
                      <span
                        class="MuiBox-root MuiBox-root-54 makeStyles-tag-32"
                      >
                        #Front-end
                      </span>
                    </div>
                  </div>
                  <img
                    src="https://picsum.photos/401/401"
                  />
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-middle"
                />
              </div>
              <div>
                <div
                  class="MuiBox-root MuiBox-root-55 makeStyles-container-35"
                >
                  <div
                    class="MuiBox-root MuiBox-root-56 makeStyles-titleContiner-36"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-57 makeStyles-title-37"
                    >
                      No Jira, No Cry
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-58 makeStyles-createdAt-39"
                    >
                      Dec 24, 2020
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-59 makeStyles-tags-40"
                    >
                      <span
                        class="MuiBox-root MuiBox-root-60 makeStyles-tag-32"
                      >
                        #Tools
                      </span>
                    </div>
                  </div>
                  <img
                    src="https://picsum.photos/402/402"
                  />
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-middle"
                />
              </div>
              <div>
                <div
                  class="MuiBox-root MuiBox-root-61 makeStyles-container-35"
                >
                  <div
                    class="MuiBox-root MuiBox-root-62 makeStyles-titleContiner-36"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-63 makeStyles-title-37"
                    >
                      [React] 02. createElement와 Component,render의 동작 원리
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-64 makeStyles-content-38"
                    >
                      Hi! I want to discuss with you how to manage Frontend architecture in large organizations. It feels to me that there are not many articles about this topic and it is not explained well.
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-65 makeStyles-createdAt-39"
                    >
                      Oct 14, 2020
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-66 makeStyles-tags-40"
                    >
                      <span
                        class="MuiBox-root MuiBox-root-67 makeStyles-tag-32"
                      >
                        #React
                      </span>
                      <span
                        class="MuiBox-root MuiBox-root-68 makeStyles-tag-32"
                      >
                        #Front-end
                      </span>
                    </div>
                  </div>
                  <img
                    src="https://picsum.photos/400/400"
                  />
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-middle"
                />
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-posts-16 MuiGrid-item MuiGrid-grid-xs-12"
            >
              <h2
                class="MuiTypography-root makeStyles-title-18 makeStyles-title-69 MuiTypography-h2"
                style="padding-left: 0px;"
              >
                Front End
              </h2>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
                >
                  <div
                    class="MuiBox-root MuiBox-root-78 makeStyles-container-70"
                  >
                    <img
                      src="https://picsum.photos/403/403"
                    />
                    <div
                      class="MuiBox-root MuiBox-root-79 makeStyles-titleContiner-71"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-80 makeStyles-title-72"
                      >
                        [React Props Cheatsheet: 10 Patterns You Should Know
                      </div>
                      <div
                        class="MuiBox-root MuiBox-root-81 makeStyles-tags-77"
                      >
                        <span
                          class="MuiBox-root MuiBox-root-82 makeStyles-tag-32"
                        >
                          #React
                        </span>
                        <span
                          class="MuiBox-root MuiBox-root-83 makeStyles-tag-32"
                        >
                          #Front-end
                        </span>
                      </div>
                      <div
                        class="MuiBox-root MuiBox-root-84 makeStyles-createdAt-74"
                      >
                        <span
                          class="MuiBox-root MuiBox-root-85 makeStyles-day-75"
                        >
                          14
                        </span>
                        <span
                          class="MuiBox-root MuiBox-root-86 makeStyles-restDate-76"
                        >
                          Mar, 2021
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
                >
                  <div
                    class="MuiBox-root MuiBox-root-87 makeStyles-container-70"
                  >
                    <img
                      src="https://picsum.photos/401/401"
                    />
                    <div
                      class="MuiBox-root MuiBox-root-88 makeStyles-titleContiner-71"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-89 makeStyles-title-72"
                      >
                        [React Props Cheatsheet: 10 Patterns You Should Know
                      </div>
                      <div
                        class="MuiBox-root MuiBox-root-90 makeStyles-tags-77"
                      >
                        <span
                          class="MuiBox-root MuiBox-root-91 makeStyles-tag-32"
                        >
                          #React
                        </span>
                        <span
                          class="MuiBox-root MuiBox-root-92 makeStyles-tag-32"
                        >
                          #Front-end
                        </span>
                      </div>
                      <div
                        class="MuiBox-root MuiBox-root-93 makeStyles-createdAt-74"
                      >
                        <span
                          class="MuiBox-root MuiBox-root-94 makeStyles-day-75"
                        >
                          11
                        </span>
                        <span
                          class="MuiBox-root MuiBox-root-95 makeStyles-restDate-76"
                        >
                          Mar, 2021
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
                >
                  <div
                    class="MuiBox-root MuiBox-root-96 makeStyles-container-70"
                  >
                    <img
                      src="https://picsum.photos/400/400"
                    />
                    <div
                      class="MuiBox-root MuiBox-root-97 makeStyles-titleContiner-71"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-98 makeStyles-title-72"
                      >
                        [React] 02. createElement와 Component,render의 동작 원리
                      </div>
                      <div
                        class="MuiBox-root MuiBox-root-99 makeStyles-content-73"
                      >
                        Hi! I want to discuss with you how to manage Frontend architecture in large organizations. It feels to me that there are not many articles about this topic and it is not explained well.
                      </div>
                      <div
                        class="MuiBox-root MuiBox-root-100 makeStyles-tags-77"
                      >
                        <span
                          class="MuiBox-root MuiBox-root-101 makeStyles-tag-32"
                        >
                          #React
                        </span>
                        <span
                          class="MuiBox-root MuiBox-root-102 makeStyles-tag-32"
                        >
                          #Front-end
                        </span>
                      </div>
                      <div
                        class="MuiBox-root MuiBox-root-103 makeStyles-createdAt-74"
                      >
                        <span
                          class="MuiBox-root MuiBox-root-104 makeStyles-day-75"
                        >
                          14
                        </span>
                        <span
                          class="MuiBox-root MuiBox-root-105 makeStyles-restDate-76"
                        >
                          Oct, 2020
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
                >
                  <div
                    class="MuiBox-root MuiBox-root-106 makeStyles-container-70"
                  >
                    <img
                      src="https://picsum.photos/410/410"
                    />
                    <div
                      class="MuiBox-root MuiBox-root-107 makeStyles-titleContiner-71"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-108 makeStyles-title-72"
                      >
                        Frontend Architecture at Scale for Large Organizations
                      </div>
                      <div
                        class="MuiBox-root MuiBox-root-109 makeStyles-tags-77"
                      >
                        <span
                          class="MuiBox-root MuiBox-root-110 makeStyles-tag-32"
                        >
                          #SW-engineering
                        </span>
                        <span
                          class="MuiBox-root MuiBox-root-111 makeStyles-tag-32"
                        >
                          #Front-end
                        </span>
                      </div>
                      <div
                        class="MuiBox-root MuiBox-root-112 makeStyles-createdAt-74"
                      >
                        <span
                          class="MuiBox-root MuiBox-root-113 makeStyles-day-75"
                        >
                          11
                        </span>
                        <span
                          class="MuiBox-root MuiBox-root-114 makeStyles-restDate-76"
                        >
                          May, 2020
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
